pkgs/top-level/all-packages.nix

```diff
@@ -25148,4 +25148,6 @@ in
 
   keycard-cli = callPackage ../tools/security/keycard-cli {};
 
+  managesieve = python3Packages.callPackage ../development/python-modules/managesieve {};
```

Comment:
if it's a package it should go python-packages.nix. if it's an application you should use buildPythonPackage.
if it's because siveshell is an application, you can build it as package in python-packages.nix and alias sieveshell = toPythonApplication (managesieve), you can find examples in nixpkgs.

Reply:
Thanks for the helpful advice